src/ucp/core/ucp_request.inl

```diff
@@ -390,11 +390,11 @@ ucp_request_send_state_reset(ucp_request_t *req,
     case UCP_REQUEST_SEND_PROTO_RNDV_PUT:
     case UCP_REQUEST_SEND_PROTO_ZCOPY_AM:
     case UCP_REQUEST_SEND_PROTO_RNDV_ACK:
-        req->send.state.uct_comp.func   = comp_cb;
         req->send.state.uct_comp.count  = 0;
         req->send.state.uct_comp.status = UCS_OK;
         /* Fall through */
     case UCP_REQUEST_SEND_PROTO_BCOPY_AM:
+        req->send.state.uct_comp.func   = comp_cb;
```

Comment:
already in https://github.com/openucx/ucx/pull/6611/files#diff-96cebf253aeccf8982cf2b3a4690e72978acaca276af7ac14c721e7ce98e99adL358